src/pbt/prophecy_build_tool.py

```diff
@@ -0,0 +1,428 @@
+                        if (
+                            potential_found_job["settings"]["name"]
+                            == job_request["name"]
+                        ):
+                            found_job = potential_found_job
+                            break
+
+                    if len(found_jobs) <= 0:
+                        break
+
+                job_request = job_definition["request"]["CreateNewJobRequest"]
+                if found_job is None:
+                    print("    Creating a new job: %s" % (job_request["name"]))
+                    self.jobs_service.create_job(**job_request)
+                else:
+                    print("    Updating an existing job: %s" % (job_request["name"]))
+                    self.jobs_service.reset_job(
+                        found_job["job_id"], new_settings=job_request, version="2.1"
+                    )
+            except Exception as e:
+                print(
+                    f"\n[bold red]❌ Create/Update for job: {job_request['name']} failed with exception: {e} [/bold red]"
+                )
+                job_update_failures[job_request["name"]] = str(e)
+
+        if len(pipelines_upload_failures) == 0 and len(job_update_failures) == 0:
+            print("\n[bold blue]✅ Deployment completed successfully![/bold blue]")
+        else:
+            print("\n[bold red]❌ Deployment failed![/bold red]")
+            if len(pipelines_upload_failures) > 0:
+                print(
+                    "   Upload failed for pipelines: %s"
+                    % (" ,".join(pipelines_upload_failures.keys()))
+                )
+            if len(job_update_failures) > 0:
+                print(
+                    "   Create/Update failed for jobs: %s"
+                    % (" ,".join(job_update_failures.keys()))
+                )
+            sys.exit(1)
+
+    def test(self):
+        if self._verify_unit_test_env():
+            unit_test_results = {}
+
+            for pipeline_i, (path_pipeline, pipeline) in enumerate(
+                self.pipelines.items()
+            ):
+                print(
+                    "\n  Unit Testing pipeline %s [%s/%s]"
+                    % (path_pipeline, pipeline_i + 1, self.pipelines_count)
+                )
+
+                path_pipeline_absolute = os.path.join(
+                    os.path.join(self.path_root, path_pipeline), "code"
+                )
+                if self.project_language == "python":
+                    if os.path.isfile(
+                        os.path.join(path_pipeline_absolute, "test/TestSuite.py")
+                    ):
+                        unit_test_results[path_pipeline] = self.test_python(
+                            path_pipeline_absolute
+                        )
+                elif self.project_language == "scala":
+                    unit_test_results[path_pipeline] = self.test_scala(
+                        path_pipeline_absolute
+                    )
```

Comment:
--do--
else block

Reply:
Same as above, already taken care of during parse project